cli/internal/run/run.go

```diff
@@ -429,6 +435,47 @@ func (r *run) run(ctx gocontext.Context, targets []string, executionState *turbo
 	)
 }
 
+func checkPackageInputsHashes(executionState *turbostate.ExecutionState, taskHashTracker *taskhash.Tracker) error {
```

Comment:
Could we leverage [`cmp.Equal`](https://pkg.go.dev/github.com/google/go-cmp/cmp#Equal) instead of the hand rolled equality check?